### PR TITLE
feat: backend support

### LINKED ISSUE #10: Backend communication
As a developer, I want to be able to make calls to the backend.

The boilerplate is already implemented in https://github.com/liftedinit/many-js/

### PR DESCRIPTION
**Pull request by fmorency**
This PR adds backend support for the `dweb` project. 

I still have to clean up the code and add some documentation but I think you can start playing with it @nblenke.

By the way, I'm happy to learn your tips and tricks on React and frontend engineering as I'm new to this stuff.

Backend to use: https://github.com/liftedinit/many-rs/pull/403

Closes #10
Uses https://github.com/liftedinit/many-js/pull/114

### CODE REVIEW

**Comment by fmorency** [comment]
@nblenke I noticed that I'm not able to deploy "bigger" websites (e.g., Alberto is 4.2MB) using the front end. However, it works when using the CLI.

The backend reports
```
2023-09-13T20:26:11.566267Z ERROR many_server::transport::http: Error decoding envelope. Error description="decode CBOR failure: Io(EndOfFile)"
```

I suspect I'm not receiving the whole envelope from the other end.

I don't see much in the Chrome console...
```
Failed to load resource: the server responded with a status of 500 (Internal Server Error)
index.js:1191 Error: Insufficient data
    at Object.decodeFirstSync (decoder.js:233:1)
    at CoseMessage.fromCborData (cose.js:33:1)
    at Message.fromCborData (index.js:70:1)
    at Network.send (network.js:31:1)
    at async Network.call (network.js:51:1)
    at async Network.deploy (web.js:24:1)
    at async Object.mutationFn (queries.ts:40:1)
```

Any idea?

**Comment by fmorency** [comment]
Might be worth investigating: https://github.com/hildjj/node-cbor/issues/170

**Comment by fmorency** [comment]
> I was able to fix this issue in my branches. I am now able to deploy 'Alberto'.

Glad you found the issue! I did notice that dropzone, the package we're using for the file drop, does have a default [maxFilesize](https://docs.dropzone.dev/configuration/basics/configuration-options) of 255 bytes, that would've been my guess.  Side note that there's other fancy things we can do there like show upload progress.

**Comment by fmorency** [comment]
I still want to do more refactoring and add proper error handling and tests.

All features should be working now.

File: src/features/network/store.ts [hunk 1]
```diff
@@ -8,9 +8,10 @@ const devDomains = ['localhost', 'liftedinit.tech']
 
 const initialState = {
   activeId: 0,
-  nextId: 1,
+  nextId: 2,
   byId: new Map([
-      [0, { name: 'Ghostcloud Production', url: '/api', filter: 'ghostcloud'}],
+    [0, { name: 'Localhost', url: 'http://localhost:8000' }],
+    [1, { name: 'Ghostcloud Production', url: '/api', filter: 'ghostcloud' }],
```
Comment (jgryffindor) [inline src/features/network/store.ts]: For some reason on https://dev.ghostcloud.org when I use web3auth I'm still getting localhost:8000 as the selected network. I added some logging to the current build deployed to capture the networks: 
```
activeNetwork:{"id":0,"name":"Localhost","url":"http://localhost:8000"}
```

Comment (jgryffindor) [reply src/features/network/store.ts]: Looks like without the dropdown we aren't yet calling setActiveId as that typically happens on page load. Gwen has this feature already for us to reference.

Comment (fmorency) [reply src/features/network/store.ts]: Yes, my bad, I will push the fix removing the dev server.

Comment (fmorency) [reply src/features/network/store.ts]: @jgryffindor can you try now?

**Comment by fmorency** [comment]
Aight this PR is becoming a bit huge; please review it now. Thanks!

**Pull request closed without merging**